Samples~/AvatarCreatorSamples/Scripts/UI/SelectionScreens/AvatarCreatorSelection.cs

```diff
@@ -301,8 +301,9 @@ private void ProcessAvatar(GameObject avatar)
                 avatar.GetComponent<Animator>().runtimeAnimatorController = animator;
             }
             avatar.transform.rotation = lastRotation;
-            avatar.AddComponent<MouseInput>();
-            avatar.AddComponent<AvatarMouseRotation>();
+            avatar.AddComponent<MouseRotationHandler>();
+            var rotator = avatar.AddComponent<AvatarRotator>();
+            rotator.SetAvatar(avatar);
```

Comment:
doesn't this feels redundant if component is attached to avatar, it will have access to avatar then why rotatar needs to be set.